spec/reservation_mgr_spec.rb

```diff
@@ -0,0 +1,157 @@
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      expect{@new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))}.must_raise ArgumentError
+    end
+    it "will create two new reservations if someone wants two rooms" do
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"),rooms: 2)
+      expect(@new_ResMgr.reservations.length).must_equal 2
+    end
+    it "will raise an argument error if not enough rooms are available" do
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      expect{@new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"),rooms: 2)}.must_raise ArgumentError
+    end
+    it "will not reserve a room on the same day if the dates are inside the date range of a previous reservation" do
+      @new_ResMgr.make_reservation(Date.parse("2018-09-04"),Date.parse("2018-09-08"))
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      expect(@new_ResMgr.reservations.length).must_equal 2
+      expect(@new_ResMgr.rooms[0].unavailable_dates.length).must_equal 1
+      expect(@new_ResMgr.rooms[1].unavailable_dates.length).must_equal 1
+    end
+    it "will add two instances of res to one room if they are not on the same date" do
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      date_hash = {check_in: Date.parse("2018-09-05"), check_out: Date.parse("2018-09-07")}
+      expect(@new_ResMgr.rooms[0].unavailable_dates[0]).must_equal date_hash
+
+      @new_ResMgr.make_reservation(Date.parse("2018-09-07"),Date.parse("2018-09-09"))
+      date_hash2 = {check_in: Date.parse("2018-09-07"), check_out: Date.parse("2018-09-09")}
+      expect(@new_ResMgr.rooms[0].unavailable_dates[1]).must_equal date_hash2
+      expect(@new_ResMgr.rooms[0].unavailable_dates.length).must_equal 2
+    end
+    it "will reserve a block room that is available if someone puts in a block party" do
+      @new_ResMgr.reserve_block(Date.parse("2018-09-05"),Date.parse("2018-09-07"),2,"Metzner")
+      res = @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"),block_id: "Metzner")
+      expect(res[0].block_available).must_equal false
+    end
+    it "will raise an argument error if the block room checkin checkout dates don't match what they want to reserve" do
+      @new_ResMgr.reserve_block(Date.parse("2018-09-05"),Date.parse("2018-09-07"),2,"Metzner")
+      expect{@new_ResMgr.make_reservation(Date.parse("2018-09-06"),Date.parse("2018-09-07"),block_id: "Metzner")}.must_raise ArgumentError
+    end
```

Comment:
You should also test what happens when you try to reserve more rooms in the block than are in that block.